ees_sharepoint/full_sync_command.py

```diff
@@ -8,14 +8,35 @@
 It will attempt to sync absolutely all documents that are available in the
 third-party system and ingest them into Enterprise Search instance."""
 from .base_command import BaseCommand
-from .fetch_index import start
+from .sync_sharepoint import init_sharepoint_sync
+from .connector_queue import ConnectorQueue
+from .sync_enterprise_search import init_enterprise_search_sync
+from multiprocessing import Process
 
 
 class FullSyncCommand(BaseCommand):
+    """This class start execution of fullsync feature."""
+
     def execute(self):
+        """This function execute the start function."""
         config = self.config
         logger = self.logger
-        workplace_search_client = self.workplace_search_client
-        sharepoint_client = self.sharepoint_client
+        args = self.args
+
+        queue = ConnectorQueue()
+        producer = Process(
```

Comment:
Once you have a queue, producers and consumers are threads or processes, not just one.

The pattern to implement is in pseudo-code:

```
queue = Queue()

def producer():
    queue.put("data_from_3rd_party")

def consumer():
    batch = queue.get_100_lines()
    send_data_to_ws(batch)

producers = Pool(10)
consumers = Pool(10)

producers.start()
consumers.start()
```

The code that is fetching the data, and the code that is sending it to WS should have zero code related to threads or processes. It puts &  grabs data from a queue.

In an ideal world your code should be separated into two distinct parts (simple names here just to explain) :

- jobs.py: the queue code, and a function to start a pool of workers given a function
- fetch.py: the function that fetches the data from sharepoint
- sync.py : the function that sends it to WS 

The two last modules here should not know what a thread or a process is